pkg/loader/loader.go

```diff
@@ -399,3 +434,42 @@ func input(ctx context.Context, cache *cache.Client, base *source, name string)
 
 	return nil, fmt.Errorf("can not load tools path=%s name=%s", base.Path, name)
 }
+
+// isOpenAPI checks if the data is an OpenAPI definition and returns the version if it is.
+func isOpenAPI(data []byte) (int, bool) {
```

Comment:
You can simplify the signature of this function. Returning `0` or `-1` to mean "not OpenAPI" seems valid. It would also make the above if-and-switch statement a little simpler.

Reply:
Makes sense, now that it's only used once